File: 卡码/代码随想录/JS/动态规划/02-背包问题/02-完全背包/LeetCode_0377.js
//377. 组合总和 Ⅳ
const combinationSum4 = function(nums, target) {
	const dp=new Array(target+1).fill(0);
	dp[0]=1;
	for(let i=0;i<=target;i++){
		for(let num of nums){
			if(i>=num) dp[i]+=dp[i-num];
		}
	}
	return dp[target];
};

File: 卡码/代码随想录/Python/动态规划/02-背包问题/02-完全背包/LeetCode_0377.py
#377. 组合总和 Ⅳ
from typing import *

class Solution:
  def combinationSum4(self, nums: List[int], target: int) -> int:
    dp:List[int]=[0]*(target+1)
    dp[0]=1
    for i in range(target+1):
      for num in nums:
        if i>=num:
          dp[i]+=dp[i-num]
    return dp[target]